### PR TITLE
Implement categories filter

### PR DESCRIPTION
**Pull request by dianashk**
Connected to pelias/pelias#150

### CODE REVIEW

**Comment by dianashk** [comment]
TBD: figure out why `reverse` was set to not handle `boundary.circle` filter. This is needed for the new `nearby` and I'd like to know the technical implications of the previous decision before enabling it for `nearby`. Important to note that it will not be enabled in `reverse`. In fact the goal of this entire PR is to preserve `reverse` as-is to the greatest extent possible.

**Comment by orangejulius** [comment]
related: https://github.com/pelias/api/issues/510

Also, I just realized, based on the [wiki page for what we want from reverse geocoding](https://github.com/pelias/pelias/wiki/Reverse-Geocoding,-Part-Deux) that we may eventually want to depricate boundary.circle. Reverse should only return things that are extremely close (a few meters) to the point specified, so a boundary radius doesn't make sense anymore. Does that sound right?

**Comment by missinglink** [comment]
@orangejulius I think you're right, the `boundary.circle` API will need to remain internally but we can remove it from our external APIs.

The original intention of that parameter was to do something like `nearby` where you might need to expand the search area (something like squash courts for instance) to get a decent list of results.

**Comment by dianashk** [comment]
I would like handle the `boundary.circle.radius` work separately and leave it unchanged in this PR. I think we will need to update the default for this param but probably can't remove it fully until v2.0.
I've assigned the incorrect warning ticket to myself and will also handle it separately.

**Comment by trescube** [comment]
:shipit: good call on splitting out parts of geojsonify to new files

File: sanitiser/_categories.js [hunk 1]
```diff
@@ -1,34 +1,53 @@
 
 var check = require('check-types');
+var categoryTaxonomy = require('pelias-categories');
+
+var ERRORS = {
```
Comment (orangejulius) [inline sanitiser/_categories.js]: I like this! Much better than having strings all over the code like we probably do in other sanitisers.

**Comment by orangejulius** [comment]
LGTM and works locally on my machine!

**Pull request merged**